.github/macos-installer/Makefile

```diff
@@ -18,6 +18,11 @@ LDFLAGS := $(TARGET_FLAGS) $(ARCH_FLAGS)
 PREFIX := /usr/local
 GIT_PREFIX := $(PREFIX)/git
 
+# Replace -rc with .rc in the version string
+# This is to ensure compatibility with the format as generated by GIT-VERSION-GEN
+ORIGINAL_VERSION := $(VERSION)
+VERSION := $(shell echo $(ORIGINAL_VERSION) | sed 's/-rc/.rc/g')
```

Comment:
A slightly more canonical form would be `VERSION := $(subst -rc,.rc,$(VERSION))`, I think. But I'll not hold up this PR over that.

Reply:
Is this a `bash`-ism? Bash on macOS is an older version, and the default shell is now `zsh` for new local accounts. I just want to make sure it's portable.

Reply:
Heh, this is `Makefile` syntax, not shell... And Git is _very_ reliant on GNU make, I don't think that it will work with any old BSD make. In particular, `$(subst ...)` is already used, my browser counts a whopping 57 matches in the top-level `Makefile`, [here is the first one](https://github.com/git/git/blob/v2.51.0/Makefile#L1506).